code/modules/research/xenobiology/xenobiology.dm

```diff
@@ -658,29 +658,28 @@
 	icon = 'icons/obj/medical/chemical.dmi'
 	icon_state = "potsilver"
 
-/obj/item/slimepotion/slime/docility/attack(mob/living/simple_animal/slime/M, mob/user)
-	if(!isslime(M))
+/obj/item/slimepotion/slime/docility/attack(mob/living/basic/slime/target_slime, mob/user)
+	if(!isslime(target_slime))
 		to_chat(user, span_warning("The potion only works on slimes!"))
 		return ..()
-	if(M.stat)
+	if(target_slime.stat)
 		to_chat(user, span_warning("The slime is dead!"))
 		return
-	if(M.rabid) //Stops being rabid, but doesn't become truly docile.
-		to_chat(M, span_warning("You absorb the potion, and your rabid hunger finally settles to a normal desire to feed."))
+	if(target_slime.ai_controller?.clear_blackboard_key(BB_SLIME_RABID)) //Stops being rabid, but doesn't become truly docile.
+		to_chat(target_slime, span_warning("You absorb the potion, and your rabid hunger finally settles to a normal desire to feed."))
 		to_chat(user, span_notice("You feed the slime the potion, calming its rabid rage."))
-		M.rabid = FALSE
+		//target_slime.ai_controller?.black
```

Comment:
delete this line there's no point having commented out code

Reply:
Thank you for pointing it out, the extract almost half did not work :pensive: